### PR TITLE
feat(change_detection): make INTERPOLATE_REGEXP customizable

### LINKED ISSUE #7305: Request: Compatible API to `$interpolateProvider.startSymbol`, `endSymbol`
As far as I know, any API that is compatible to `$interpolateProvider.startSymbol()` of ng1 does not exist.
In ng2, is this use case deprecated? Or it's just currently missing?

cc: @IgorMinar

### PR DESCRIPTION
**Pull request by lacolaco**
- **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  Feature
- **What is the current behavior?** (You can also link to an open issue here)
  The pattern for interpolation is not able to modify. `INTERPOLATE_REGEXP` is a variable and is referred in `Parser` directly.
- **What is the new behavior (if this is a feature change)?**
  Add new public api, `INTERPOLATE_REGEXP`, which allows us customize interpolation regexp pattern. (for Python Jinja and more template languages). 
  Similar api is in AngularJS as `$interpolateProvider`.

And `Parser` takes the pattern for interpolation from DI. 
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
  `Parser` constructor required new parameter `interpolateRegexp` as second argument.
- **Other information**:
  Close #7305

### CODE REVIEW

File: modules/angular2/src/core/change_detection/parser/parser.ts [hunk 1]
```diff
@@ -63,7 +62,8 @@ export class Parser {
   _reflector: Reflector;
 
   constructor(/** @internal */
-              public _lexer: Lexer, providedReflector: Reflector = null) {
+              public _lexer: Lexer, @Inject(INTERPOLATE_REGEXP) public _interpolateRegex: RegExp,
```
Comment (tbosch) [inline modules/angular2/src/core/change_detection/parser/parser.ts]: The compiler should not be configured via DI but rather via a new field on `CompilerConfig` (a new class that replaces `ChangeDetectorConfig` once you rebase).

**Comment by tbosch** [comment]
Need a rebase as well.

**Comment by laco0416** [comment]
I've rebased but then new feature's tests were dropped. 
I'll add tests.

**Comment by laco0416** [comment]
New approach: `CompilerConfig.interpolateRegexp` as new constructor optional argument.

```ts
new CompilerConfig(true, true, true, null, /<<([\s\S]*?)>>/g)
```

**Pull request merged**